### DISCUSSION TITLE
Learning rate and convergence threshold for cudaq's Adam optimizer

### DISCUSSION THREAD

**Discussion by anandaroy7** [NVIDIA/cuda-quantum#2329] [General]
Hello, Is it possible for one to tune the learning rate and the convergence threshold for the Adam optimizer? 

Thanks.

**Comment by bmhowe23** [comment]
Hi @anandaroy7 - are you using C++ or Python? If you're using C++, I think you can use the `f_tol` and `step_size` attributes. I am not sure if these are exposed to Python at the moment, but if they are missing, we could certainly add them.

**Reply by anandaroy7** [reply]
Hi, I am using the python version and it is not practical for me to switch to C++. It would be great if you could add the ability to control the convergence threshold and the learning rate in the python version. Thanks!
